VocalTractLabApi.cpp

```diff
@@ -1288,6 +1288,57 @@ int vtlGesturalScoreToTractSequence(const char* gesFileName, const char* tractSe
 }
 
 
+
+int vtlGetGesturalScoreDuration(const char* gesFileName, int* audioFileDuration, int* gesFileDuration)
+{
+    if (!vtlApiInitialized)
+    {
+        printf("Error: The API has not been initialized.\n");
+        return 1;
+    }
+
+
+
+    // ****************************************************************
+    // Init and load the gestural score.
+    // ****************************************************************
+
+    GesturalScore* gesturalScore = new GesturalScore(vocalTract, glottis[selectedGlottis]);
+    static const int NUM_CHUNCK_SAMPLES = 110;
+
+    bool allValuesInRange = true;
+    if (gesturalScore->loadGesturesXml(string(gesFileName), allValuesInRange) == false)
+    {
+        printf("Error in vtlGesturalScoreToGlottisSignals: Loading the gestural score file failed!\n");
+        delete gesturalScore;
+        return 2;
+    }
+
+    if (allValuesInRange == false)
+    {
+        printf("Error in vtlGesturalScoreToGlottisSignals: Some values in the gestural score are out of range!\n");
+        delete gesturalScore;
+        return 3;
+    }
+
+    // Important !!!
+    gesturalScore->calcCurves();
+
+    if (gesFileDuration != NULL)
+    {
+        *gesFileDuration = gesturalScore->getDuration_pt();
+    }
+
+    if (audioFileDuration != NULL)
+    {
+        *audioFileDuration = (int( ( gesturalScore->getDuration_pt() ) / NUM_CHUNCK_SAMPLES ) + 1)  * NUM_CHUNCK_SAMPLES;
```

Comment:
The C-style cast should be  ``(int)`` (surrounded by parentheses). This may fail on some compilers.

Reply:
Done.